### PR TITLE
Protocol TVL, pool count, and various other updates

### PR DESCRIPTION
**Pull request by jmulq**
- Update to protocol TVL
- Add protocol pool count entity updates
- Fix underlying exchange event issue
- Separate underlying vs normal exchange logic
- Handle Curve Fi Zap deposits
- Add entity key manager and refactor graph out to support new manager

### CODE REVIEW

File: src/constants.rs [hunk 1]
```diff
@@ -105,16 +105,25 @@ pub static BLACKLISTED_TOKENS: &[&str] = &[
...
 
-pub fn default_usd_price() -> BigDecimal {
-    BigDecimal::from(1000000)
+pub fn default_usdc_decimals() -> u64 {
+    6
+}
+
+// Price per USDC in token's units
+pub fn usdc_smallest_unit_value() -> BigDecimal {
+    BigDecimal::from(10u64.pow(default_usdc_decimals() as u32))
```
Comment (Dirt-Nasty) [inline src/constants.rs]: Just a reminder that BigDecimal and BigInt has built in decimal handlers you might want to check out.

https://docs.rs/substreams/latest/substreams/scalar/struct.BigInt.html#method.to_decimal
https://docs.rs/substreams/latest/substreams/scalar/struct.BigDecimal.html#method.divide_by_decimals

File: src/modules/420_graph_out.rs [hunk 2]
```diff
@@ -454,3 +497,62 @@ fn create_swap_entity(tables: &mut Tables, event: &PoolEvent, swap: &SwapEvent)
         .set("amountOutUSD", BigDecimal::zero())
         .set("pool", format::format_address_string(&event.pool_address));
 }
+
+fn create_swap_underlying_entity(
+    tables: &mut Tables,
+    event: &PoolEvent,
+    swap_underlying: &SwapUnderlyingEvent,
+) {
+    let key = format!("swap-0x{}-{}", event.transaction_hash, event.log_index);
```
Comment (Dirt-Nasty) [inline src/modules/420_graph_out.rs]: key manager?

File: src/modules/420_graph_out.rs [hunk 3]
```diff
@@ -454,3 +497,62 @@ fn create_swap_entity(tables: &mut Tables, event: &PoolEvent, swap: &SwapEvent)
...
+        .create_row("Swap", key)
+        .set(
+            "hash",
+            format::format_address_string(&event.transaction_hash),
+        )
+        .set("logIndex", event.log_index as i32)
+        .set("protocol", utils::get_protocol_id())
+        .set("to", format::format_address_string(&event.to_address))
+        .set("from", format::format_address_string(&event.from_address))
+        .set("blockNumber", BigInt::from(event.block_number))
```
Comment (Dirt-Nasty) [inline src/modules/420_graph_out.rs:519]: If these are Strings you can use .set_bigint instead of .set

File: src/constants.rs (hunk 1 above)
Comment (jmulq) [reply src/constants.rs]: Removed this function for now as I don't actually need it, but will use the built in decimal handlers in future.

File: src/modules/420_graph_out.rs (hunk 2 above)
Comment (jmulq) [reply src/modules/420_graph_out.rs]: Good spot, have changed this.

File: src/modules/420_graph_out.rs (hunk 3 above)
Comment (jmulq) [reply src/modules/420_graph_out.rs:519]: They were `u64`'s so kept as set for now.

**Pull request merged**